sdks/java/harness/src/main/java/org/apache/beam/fn/harness/PTransformRunnerFactory.java

```diff
@@ -91,6 +88,18 @@
     /** Returns a {@link FnDataReceiver} to send output to for the specified PCollection id. */
     <T> FnDataReceiver<T> getPCollectionConsumer(String pCollectionId);
 
+    /**
+     * Returns all {@link Endpoints.ApiServiceDescriptor} to {@link BeamFnDataOutboundAggregator}
+     * mappings.
+     */
+    Map<Endpoints.ApiServiceDescriptor, BeamFnDataOutboundAggregator> getOutboundAggregators();
```

Comment:
Instead of exposing the aggregator, consider exposing a function like 
```
FnDataReceiver<T> receiver addOutgoingDataEndpoint(Endpoints.ApiServiceDescriptor apiServiceDescriptor, Coder<T> coder)
```
and ditto for timers with:
```
FnDataReceiver<Timer<T>> receiver addOutgoingTimerEndpoint(String timerFamilyId, Coder<Timer<T>> coder)
```

this will make testing the `BeamFnDataWriteRunner` and `FnApiDoFnRunner` simpler.

Reply:
done.